src/script/api/script_map.cpp

```diff
@@ -18,57 +18,57 @@
 	return ::IsValidTile(t);
 }
 
-/* static */ TileIndex ScriptMap::GetMapSize()
+/* static */ SQInteger ScriptMap::GetMapSize()
 {
 	return ::Map::Size();
 }
 
-/* static */ uint32 ScriptMap::GetMapSizeX()
+/* static */ SQInteger ScriptMap::GetMapSizeX()
 {
 	return ::Map::SizeX();
 }
 
-/* static */ uint32 ScriptMap::GetMapSizeY()
+/* static */ SQInteger ScriptMap::GetMapSizeY()
 {
 	return ::Map::SizeY();
 }
 
-/* static */ int32 ScriptMap::GetTileX(TileIndex t)
+/* static */ SQInteger ScriptMap::GetTileX(TileIndex t)
 {
 	if (!::IsValidTile(t)) return -1;
 	return ::TileX(t);
 }
 
-/* static */ int32 ScriptMap::GetTileY(TileIndex t)
+/* static */ SQInteger ScriptMap::GetTileY(TileIndex t)
 {
 	if (!::IsValidTile(t)) return -1;
 	return ::TileY(t);
 }
 
-/* static */ TileIndex ScriptMap::GetTileIndex(uint32 x, uint32 y)
+/* static */ TileIndex ScriptMap::GetTileIndex(SQInteger x, SQInteger y)
 {
 	return ::TileXY(x, y);
```

Comment:
Should this do validity/precondition checks like `0 <= x && x < ScriptMap::GetMapSizeX()`?

Reply:
I wondered about this one, at first I added validation and that needed an update of regression expected output.
Then looking at the original expected output I though it was intentional to not validate the index.
It allows to do things like `tile + AIMap.GetTileIndex(-1, -1)`

Reply:
Then the documentation of that function needs to be amended. Something along the lines of: when 0 <= x && x < SizeX() && 0 <= y && y < SizeY(), then a valid tile index is returned. Otherwise it may be invalid, but could be used to calculated neighbouring tiles, e.g. tile + AIMap.GetTileIndex(-1, -1) gets the tile index of the tile to the north. But be aware that even when tile is a valid tile, the result might not be a valid tile.